bitnami/prometheus-operator/templates/prometheus/prometheus.yaml

```diff
@@ -177,7 +177,7 @@ spec:
           protocol: TCP
     {{- end }}
     {{- if .Values.prometheus.containers }}
-    {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | indent 4 }}
+    {{ toYaml .Values.prometheus.containers | indent 4 }}
```

Comment:
The `"prometheus-operator.tplValue"` macro should detect that `.Values.prometheus.containers` is a YAML and render it accordingly. See:
```
{{/*
Renders a value that contains template.
Usage:
{{ include "prometheus-operator.tplValue" ( dict "value" .Values.path.to.the.Value "context" $) }}
*/}}
{{- define "prometheus-operator.tplValue" -}}
    {{- if typeIs "string" .value }}
        {{- tpl .value .context }}
    {{- else }}
        {{- tpl (.value | toYaml) .context }}
    {{- end }}
{{- end -}}
```
Replacing `indent` with `nindent` should be enough:
```diff
-     {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | indent 4 }}
+     {{- include "prometheus-operator.tplValue" (dict "value" .Values.prometheus.containers "context" $) | nindent 4 }}
```